From 30f39442241d92fe00e71654b0ddcebd65fd57dc Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Wed, 8 Nov 2023 15:59:29 +0530
Subject: [PATCH] mtd: nand: add TC58NYG0S3HBAI4, TH58NYG2S3HBAI4 and
 TH58NYG3S0H

Add TC58NYG0S3HBAI4, TH58NYG2S3HBAI4 and TH58NYG3S0H to nand_ids table.

Change-Id: I33e58c1f43fdb374d30c9b27025a54e84f3e1bb4
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/mtd/nand/raw/nand_ids.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mtd/nand/raw/nand_ids.c b/drivers/mtd/nand/raw/nand_ids.c
index 1b4fdbd87f70..5b6de79b7102 100644
--- a/drivers/mtd/nand/raw/nand_ids.c
+++ b/drivers/mtd/nand/raw/nand_ids.c
@@ -29,6 +29,12 @@ struct nand_flash_dev nand_flash_ids[] = {
 	{"TC58NVG0S3E 1G 3.3V 8-bit",
 		{ .id = {0x98, 0xd1, 0x90, 0x15, 0x76, 0x14, 0x01, 0x00} },
 		  SZ_2K, SZ_128, SZ_128K, 0, 8, 64, NAND_ECC_INFO(1, SZ_512), },
+	{"TC58NYG0S3HBAI4 1.8V 1Gbit",
+		{ .id = {0x98, 0xa1, 0x80, 0x15} },
+		SZ_2K, SZ_128, SZ_128K, 0, 4, 128, NAND_ECC_INFO(8, SZ_512) },
+	{"TH58NYG2S3HBAI4 1.8V 8-bit",
+                { .id = {0x98, 0xac, 0x91, 0x15} },
+                  SZ_2K, SZ_512, SZ_128K, 0, 4, 128, NAND_ECC_INFO(8, SZ_512) },
 	{"TC58NVG2S0F 4G 3.3V 8-bit",
 		{ .id = {0x98, 0xdc, 0x90, 0x26, 0x76, 0x15, 0x01, 0x08} },
 		  SZ_4K, SZ_512, SZ_256K, 0, 8, 224, NAND_ECC_INFO(4, SZ_512) },
@@ -138,6 +144,10 @@ struct nand_flash_dev nand_flash_ids[] = {
 		{ .id = {0xc2, 0x94} },
 		SZ_2K, SZ_128, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
 
+	{"TH58NYG3S0H 8G 1.8V 8-bit",
+		{ .id = {0x98, 0xa3, 0x91, 0x26} },
+		SZ_4K, SZ_1K, SZ_256K, 0, 4, 256, NAND_ECC_INFO(8, SZ_512)},
+
 	LEGACY_ID_NAND("NAND 4MiB 5V 8-bit",   0x6B, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE3, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE5, 4, SZ_8K, SP_OPTIONS),
-- 
2.34.1

